packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,145 @@
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+interface RelayAncillaryData {
+  depositId: number;
+  depositTimestamp: number;
+  recipient: string;
+  l2Sender: string;
+  l1Token: string;
+  amount: number;
+  slowRelayFeePct: number;
+  instantRelayFeePct: number;
+  quoteTimestamp: number;
+  realizedLpFeePct: number;
+  depositContract: string;
+  priceRequestTime: string;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond "Yes" or "No"
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly toBNWei: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+    this.toBNWei = (_number) => toBN(toWei(_number.toString()));
+  }
+
+  // This method returns the validity of a relay price request attempt. The relay request was valid if and only if it:
+  // The price request's ancillary data contains parameters that match with an L2 deposit event.
+  public async getHistoricalPrice(time: number | string, ancillaryData: string): Promise<BN> {
+    // Note: `time` is unused in this method because it is not included in the relay ancillary data.
+
+    // Parse ancillary data for relay request and find deposit if possible with matching params.
+    const parsedAncillaryData: RelayAncillaryData = (parseAncillaryData(
+      ancillaryData
+    ) as unknown) as RelayAncillaryData;
+    const matchedDeposit = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === parsedAncillaryData.depositId &&
+        deposit.timestamp === parsedAncillaryData.depositTimestamp &&
+        deposit.recipient.substr(2).toLowerCase() === parsedAncillaryData.recipient &&
```

Comment:
the other option would be to cast the ancillary data part to `toChecksumAddress` and then validate the equality. I think what you have here is _ok_ but curious what you think

Reply:
yeah I like this its a bit more readable. will change.